pkg/porter/build_test.go

```diff
@@ -233,3 +233,46 @@ func TestPorter_paramRequired(t *testing.T) {
 	require.True(t, ok)
 	require.True(t, p2.Required)
 }
+
+func TestPorter_generateImages(t *testing.T) {
```

Comment:
Do we want test(s) that vet the required fields when defining an ImageMap in a porter.yaml vs fields that can be empty (`omitempty`)?  Or would this more be testing general yaml -> golang struct flow and thus perhaps not needed?

Reply:
Sorry, I'm not quite sure what you want to test here? Do you mean like the kind of test that I did for cnab-go here? To ensure that the proper set of fields have omit empty and force people to think about it each time a field is added?

https://github.com/deislabs/cnab-go/pull/24

Reply:
Oh yup, something similar to what you've done in the link provided.  Can be a follow-up if we deem it necessary.

Reply:
Yeah I can do that in a follow-up PR since it's a bit larger in scope than just image map, and we can adjust which ones do have omit empty on them at the same time.

Reply:
I am not sure I'm going to get to this today so here's an issue to track it https://github.com/deislabs/porter/issues/425